vm/src/memory/call_stack.cpp

```diff
@@ -115,3 +115,18 @@ size_t CallStack::size() const {
 bool CallStack::is_empty() const {
     return size() == 0;
 }
+std::vector<stack_frame_view> CallStack::dump_stack() const {
```

Comment:
The stack must be unwound, i.e. all the frames should be removed. This reduces the need of using the `SIZE_MAX` constant trick, and would restore a sane VM state.